spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/DefaultDatastoreEntityConverter.java

```diff
@@ -62,6 +67,27 @@ public ReadWriteConversions getConversions() {
 		return this.conversions;
 	}
 
+	@Override
+	public <R> Map<String, R> readAsMap(TypeInformation<R> type, BaseEntity entity,
+			boolean embeddedMapValueIsEmbedded) {
+		if (entity == null) {
+			return null;
+		}
+		Map<String, R> result = new HashMap<>();
+		EntityPropertyValueProvider propertyValueProvider = new EntityPropertyValueProvider(
+				entity, this.conversions);
+		Set<String> fieldNames = entity.getNames();
+		for (String field : fieldNames) {
+			result.put(field,
+					propertyValueProvider.getPropertyValue(field,
+							embeddedMapValueIsEmbedded, false,
```

Comment:
Right now this is always false to mean no arbitrary level of embedded value types. but we can put an integer arg in the embedded annotation as that setting and remove this limitation.

Reply:
no longer necessary since i am now using the entity annotation on the embedded object's class.